app/controllers/submissions_controller.rb

```diff
@@ -659,9 +659,23 @@ def update_submissions
       changed = if assignment.is_peer_review?
                   set_pr_release_on_results(groupings, release)
                 else
-                  set_release_on_results(groupings, release)
+                  begin
+                    Result.set_release_on_results(groupings, release)
+                  rescue StandardError => e
+                    error_message, group_names = e.message.split
+                    if error_message == 'no_submission'
+                      flash_now(:error, I18n.t('submissions.errors.no_submission', group_name: group_names))
```

Comment:
You can use `I18n.t` inside model methods. This means that you can raise an exception with the exact message you want to display, and then do something like a `flash_now(:error, e.message)` here in the controller.

Reply:
Oh, and in order to make `changed = 0`, I believe you can just put in a `0` at the end of the `rescue` block, and then the result of "evaluating" that block will be `0`, which gets assigned to `changed`.